src/OrleansTestingHost/Extensions/TestConfigurationExtensions.cs

```diff
@@ -14,37 +14,40 @@ public static class TestConfigurationExtensions
         /// <summary>
         /// This call tweaks the cluster config with settings specific to a test run.
         /// </summary>
-        public static void AdjustForTestEnvironment(this ClusterConfiguration clusterConfig)
+        public static void AdjustForTestEnvironment(this ClusterConfiguration clusterConfig, string dataConnectionStringFallback)
```

Comment:
We can remove this method once we deprecate `TestingSiloHost`. For now it's code that is shared between `TestClusterOptions` and `TestingSiloHost`. It won't grow in parameter count with more overrides, as we are adding setting more values in the ClusterConfiguration and ClientConfiguration objects directly, and we are less and less relying on this kind of auto-magic replacements (which were most practical with XML configuration, but not that needed with programmatic)